Source/JavaScriptCore/runtime/PropertyTable.h

```diff
@@ -238,182 +213,240 @@ class PropertyTable final : public JSCell {
     template<typename T>
     static T* skipDeletedEntries(T* valuePtr, T* endValuePtr);
 
-    // The table of values lies after the hash index.
-    ValueType* table();
-    const ValueType* table() const;
-
-    ValueType* tableEnd() { return table() + usedCount(); }
-    const ValueType* tableEnd() const { return table() + usedCount(); }
-
     // total number of  used entries in the values array - by either valid entries, or deleted ones.
     unsigned usedCount() const;
 
     // The size in bytes of data needed for by the table.
-    size_t dataSize();
+    size_t dataSize(bool isCompact);
 
     // Calculates the appropriate table size (rounds up to a power of two).
     static unsigned sizeForCapacity(unsigned capacity);
 
     // Check if capacity is available.
-    bool canInsert();
+    bool canInsert(const ValueType&);
+
+    void remove(VM&, KeyType, unsigned entryIndex, unsigned index);
+    std::tuple<unsigned, unsigned, PropertyOffset, uint8_t> find(const KeyType&);
+
+    template<typename Index, typename Entry>
+    ALWAYS_INLINE std::tuple<unsigned, unsigned, PropertyOffset, uint8_t> findImpl(const Index*, const Entry*, const KeyType&);
+
+    bool isCompact() const { return m_indexVector & isCompactFlag; }
+
+    template<typename Functor>
+    void forEachPropertyMutable(const Functor&);
+
+    // The table of values lies after the hash index.
+    CompactPropertyTableEntry* tableFromIndexVector(uint8_t* index)
+    {
+        return reinterpret_cast_ptr<CompactPropertyTableEntry*>(index + m_indexSize);
```

Comment:
nit: Let's use `bitwise_cast` for these functions

Reply:
Done.